### ISSUE TITLE
Problem to run Cortx-motr cluster with S3

### ISSUE THREAD

**Issue by lia54** [Seagate/cortx-s3server#1361]
I was trying to run a Cortx-motr cluster with a S3 and after running the bootstrap command I am getting a failed services for S3.

Is any way in which I can know why is that teh linmotr-helpers.so is not loaded?

Oct 07 12:56:35 smc3-m03.colo.seagate.com systemd[1]: Starting Motr-S3 Core instance Server...
-- Subject: Unit s3server@0x7200000000000001:0x1f.service has begun start-up
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit s3server@0x7200000000000001:0x1f.service has begun starting up.
Oct 07 12:56:35 smc3-m03.colo.seagate.com s3startsystem.sh[22780]: + s3server --s3pidfile /var/run/s3server.0x7200000000000001:0x1f.pid --motrlocal 10.230.244.130@tcp:12345:3:1 --motrha 10.230.244.130@tcp:12345:1:1 --motrprofilefid '<0x7000000000000001:0x43>' --motrprocessfid '<0x7200000000000001:0x1f>' --s3port 28071 --log_dir /var/log/seagate/s3/s3server-0x7200000000000001:0x1f
**Oct 07 12:56:35 smc3-m03.colo.seagate.com s3startsystem.sh[22780]: s3server: error while loading shared libraries: libmotr-helpers.so.0: cannot open shared object file: No such file or directory**
Oct 07 12:56:35 smc3-m03.colo.seagate.com systemd[1]: s3server@0x7200000000000001:0x1f.service: control process exited, code=exited status=127
Oct 07 12:56:35 smc3-m03.colo.seagate.com systemd[1]: Failed to start Motr-S3 Core instance Server.
-- Subject: Unit s3server@0x7200000000000001:0x1f.service has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit s3server@0x7200000000000001:0x1f.service has failed.
-- 
-- The result is failed.
Oct 07 12:56:35 smc3-m03.colo.seagate.com systemd[1]: Unit s3server@0x7200000000000001:0x1f.service entered failed state.
Oct 07 12:56:35 smc3-m03.colo.seagate.com systemd[1]: s3server@0x7200000000000001:0x1f.service failed.

**Comment by johnbent** [comment]
Hey @lia54. Yeah, you don't have access to the Jira issue but that's fine. It is mirrored so all activity there will appear here. We just mirror it for the sake of the dev team so they can keep everything in Jira.

**Comment by lia54** [comment]
> > Rajesh Nambiar commented in Jira Server:
> 
> Not needed
> 
> hctl... commands work with the installed cortx rpm.

Is this the same rpm generated in Jenkins job mini provision seagate system? 
After having one rpm generated. How I can use it to install a cortx cluster in a fresh VM from scratch?

> 
> Normally the libraries get picked from standard localtion say /lib64/...
> 
> Please Give below command to see s3server binary is linked to which motr library
> 
> ldd /opt/seagate/cortx/s3/bin/s3server | grep motr
> 
> It should be showing output as below {code:java} [root@ssc-vm-g2-rhev4-1353 /]# ldd /opt/seagate/cortx/s3/bin/s3server | grep motr libmotr-helpers.so.0 => /lib64/libmotr-helpers.so.0 (0x00007f81b5389000) libmotr.so.2 => /lib64/libmotr.so.2 (0x00007f81b4997000) {code}  
> 
When trying this I am getting empty links as bellow:
motr libmotr-helpers.so.0 =>
 libmotr.so.2 =>

> There are chances that probably you might have tried to run some script in local cortex repo which might have resulted in copying local s3server binary to /opt/seagate/cortx/s3/bin these binaries wont be linked to motr libraries in /lib64 (say /lib64/libmotr-helpers.so.0), however when you run with script dev-start.sh, that script will define LD_LIBRARY_PATH to local third_part directory from where the libraries are getting picked dynamically.

This makes sense, but still don't know what I should do to set up properly links to the appropriate location.

**Comment by hessio** [comment]
Hi @sachitanands could you help with this issue? Or at least help us find someone who could help with this issue?

**Comment by lia54** [comment]
Hi Rajesh Nambiar @knrajnambiar76 , to setup s3-server I am using the Quick Start Guide that it is in this link:
https://github.com/Seagate/cortx-s3server/blob/main/docs/CORTX-S3%20Server%20Quick%20Start%20Guide.md

In which part of that guide, the rpm for cortx-s3server are generated and installed? Do you have another documentation I can follow for that? Thanks!

**Comment by hessio** [comment]
Hi @lia54 there is no way to do this with the current documentation. Right now you can build the s3server which will have Motr and HARE running but you cannot setup Motr and HARE and then add s3 (this might be possible but I have not seen a guide on how to do this). @sachitanands do you know if you could do this?

**Comment by lia54** [comment]
Hi @hessio thanks for the feedback! I wanted to integrate S3 server to be able to use hctl commands with hare running. I guess the key thing is how to generated link the S3 server with the libmotr library that are not the ones in the third_local directory, but the one installed with rpm from /lib64. Does anyone know what are the steps to link S3 server binaries to libmotr library in /lib64?

**Comment by hessio** [comment]
Yes sure I would like to know this too. @nileshgovande do you know if this is possible?

**Comment by hessio** [comment]
@sachitanands do you know?

**Comment by lia54** [comment]
Hi, 
Does anyone know how to get the cortx-motr cluster (motr + hare+s3) running with rpm based deployment??
